test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -0,0 +1,541 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ucx_wrapper.h"
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <string.h>
+#include <assert.h>
+
+#include <iostream>
+#include <limits>
+#include <sstream>
+
+
+struct ucx_request {
+    UcxCallback  *callback;
+    bool         completed;
+    uint32_t     conn_id;
+    ucs_status_t status;
+};
+
+
+class DebugStream : public std::ostream {
+public:
+    static DebugStream& get() {
+        if (_self == NULL) {
+            _self = new DebugStream;
+        }
+
+        return *_self;
+    }
+
+    static void destroy() {
+        if (_self != NULL) {
+            delete _self;
+        }
+    }
+
+    template<typename T>
+    DebugStream& operator<<(const T &t) {
+#if _DEBUG
+        std::cout << t;
+#endif
+        return *_self; 
+    }
+
+    typedef std::basic_ostream<char, std::char_traits<char> > EndLine;
+    typedef EndLine& (*EndLineManupulator)(EndLine&);
+    DebugStream& operator<<(EndLineManupulator manupulator) {
+#if _DEBUG
+        manupulator(std::cout);
+#endif
+        return *_self;
+    }
+
+private:
+    DebugStream() {
+    }
+
+    ~DebugStream() {
+        _self = NULL;
+    }
+
+    // singleton
+    static DebugStream *_self;
+};
+
+DebugStream* DebugStream::_self = NULL;
+
+static DebugStream& debug() {
+   return DebugStream::get() << "[DEBUG] ";
```

Comment:
i think DebugStream is overkill, can't we just use std::cout?

Reply:
debug may be useful for data path too, `std::cout` will impact performance.

Reply:
so pls try to simplify this class,